test/test_associations_middleware.js

```diff
@@ -0,0 +1,90 @@
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+/* global run */
+
+'use strict';
+
+import app           from './server';
+import db            from '../src/models/db';
+import should        from 'should';
+import supertest     from 'supertest';
+
+import * as ERR from '../src/errors';
+import {
+  entities
+} from './constants';
+
+const ERRORS = ERR.errors;
+const ERRNOS = ERR.errnos;
+
+const server = supertest.agent(app);
+
+const getError = (done, endpoint, code, errno, error) => {
+  server.get(endpoint)
+  .expect('Content-Type', /json/)
+  .expect(code)
+  .end((err, res) => {
+    should.not.exist(err);
+    res.status.should.be.equal(code);
+    res.body.code.should.be.equal(code);
+    res.body.errno.should.be.equal(errno);
+    res.body.error.should.be.equal(error);
+    done();
+  });
+}
+
+db().then(models => {
+  let noAssociations = {};
+  entities.forEach(entity => {
+    noAssociations[entity] = [];
+    const associations = models[entity].associations;
+    entities.forEach(entity_ => {
+      if (!associations[entity_] && entity !== entity_) {
+        noAssociations[entity].push(entity_);
+      }
+    });
+  });
+
+  describe('Associations', () => {
+    describe('Invalid associations', () => {
+      entities.forEach(model => {
+        noAssociations[model].forEach(anotherModel => {
+          const endpoint = '/' + model + '(1)/' + anotherModel;
+          it('GET ' + endpoint + ' should respond 400 errno 102 ' +
+             'INVALID_ASSOCIATION', done => {
+            getError(done, endpoint, 400, ERRNOS[ERR.ERRNO_INVALID_ASSOCIATION],
+                     ERRORS[ERR.BAD_REQUEST]);
+          });
+        });
+      });
+    });
+
+    describe('Valid associations but not found entity', () => {
```

Comment:
Can we test a valid association and found entity, but the entity is not linked to the previous one?
`/A(1)/B(2)/C ` where A(1) and B(2) exist, but A(1) and B(2) are not linked.